tests/non_integers_repr.rs

```diff
@@ -144,12 +144,16 @@ fn test_record2() {
 #[test]
 fn test_inline_record() {
     bin_prot_test! {
-        .. .. .. .. .. .. .. .. .. .. .. .. 0x00 0x00 0x00 -> B {
-            y: BInner { w: 0, x: 0 },
+        .. .. .. .. .. .. .. .. .. .. .. .. 0x00 0x00 0x00 0x00 0x01-> B3 {
+            y: BInner2::V1(BInner4::V0(BInner { w: 0, x: 0 })),
             z: None,
         },
-        0x00 0x7f 0xff 0xff 0xff 0xfd 0x7f 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xfc -> B {
-            y: BInner { w: 9223372036854775807, x: 2147483647 },
+        0x00 0x7f 0xff 0xff 0xff 0xfd 0x7f 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xfc 0x00 0x01 -> B2 {
+            y: BInner3 { dummy: 1 , inner: BInner2::V0(BInner { w: 9223372036854775807, x: 2147483647 })},
```

Comment:
why are these two types different? in the bin_prot tests it looks like they're the same type, but maybe I'm missing something https://github.com/janestreet/bin_prot/blob/6c45b4249fc2c3a19574c4c788e2d296cae7fc30/test/non_integers_repr.ml#L745

Reply:
On ocaml original test the type is the same, on our rust side we can represent the same type in more than one way, I explored this using two different rust types in order to showcase this. When using this in a real application you could choose what works best case by case